File: csharp/ToolGood.Algorithm/Internals/Functions/MathTrigonometric/Function_SIN.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.MathTrigonometric
{
	internal class Function_SIN : Function_1
    {
        public Function_SIN(FunctionBase[] func1) : base(func1)
        {
        }

        public override string Name => "Sin";

        public override Operand Evaluate(AlgorithmEngine work, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = GetNumber_1(work, tempParameter);
            if (args1.IsError) { return args1; }
            return Operand.Create(Math.Sin(args1.DoubleValue));
        }

    }

    

}

File: JavaScript/src/Internals/Functions/MathTrigonometric/Function_SIN.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_SIN extends Function_1 {
    get Name() {
        return "Sin";
    }

    constructor(a) {
        super(a);
    }

    Evaluate(work, tempParameter) {
        let args1 = this.GetNumber_1(work, tempParameter);
        if (args1.IsError) { return args1; }
        return Operand.Create(Math.sin(args1.DoubleValue));
    }
}

export { Function_SIN };

